File: Saams.Net/Saams.EF/UserManagement/Shift.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Saams.EF.UserManagement
{
    [Table("Shifts")]
    public class Shift : IEntity
    {
        public int Id { get; set; }

        [Required]
        public required string Name { get; set; }

        [Required]
        public required string Code { get; set; }

        [Required]
        public ShiftType Type { get; set; }

        [Required]
        public TimeSpan EntryTime { get; set; }

        [Required]
        public TimeSpan GraceEntryTime { get; set; }

        [Required]
        public TimeSpan ExitLunch { get; set; }

        [Required]
        public TimeSpan EntryLunch { get; set; }

        [Required]
        public TimeSpan ExitTime { get; set; }

        [Required]
        public TimeSpan GraceExitTime { get; set; }

        [Required]
        public TimeSpan OverTimeAllowance { get; set; }

        public ICollection<User> Users { get; set; } = new List<User>();
    }

    public enum ShiftType
    {
        ROTATIONAL,
        NON_ROTATIONAL
    }
}

File: saams-ui/src/model/shift.ts
import { Base } from "./base";
import Response from "./response"

export interface Shift extends Base {
    name: string;
    code: string;
    type: string;
    entryTime: string; // D.HH:mm:ss
    graceEntryTime: string; // D.HH:mm:ss
    exitLunch: string; // D.HH:mm:ss
    entryLunch: string; // D.HH:mm:ss
    exitTime: string; // D.HH:mm:ss
    graceExitTime: string; // D.HH:mm:ss
    overTimeAllowance: string; // D.HH:mm:ss
}

export interface ShiftResponse extends Response {
    shift: Shift;
    shifts: Shift[];
}

export const ShiftType = {
    rotational: "ROTATIONAL",
    nonrotational: "NON_ROTATIONAL"
}